solidity-its/src/main/java/org/sonarsource/solidity/its/SolidityRuling.java

```diff
@@ -0,0 +1,51 @@
+package org.sonarsource.solidity.its;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+
+public class SolidityRuling {
+
+  private static final Logger LOG = Loggers.get(SolidityRuling.class);
+
+  private SolidityRuling() {
+    // ... no reason for a public constructor
+  }
+
+  private static final String DIR = "solidity-test-sources/src/";
+  private static final String[] PROJECTS_TO_ANALYZE = {
+    "ethereum-api",
+    "Random-Files"
+  };
+
+  public static String[] getProjects() {
+    return PROJECTS_TO_ANALYZE;
+  }
+
+  public static void collectSolidityFiles() {
+    Arrays.asList(getProjects())
+      .stream()
+      .map(x -> String.format("%s%s", DIR, x))
+      .forEach(path -> {
+        try {
+          Files.find(Paths.get(path),
+            Integer.MAX_VALUE,
+            (filePath, fileAttr) -> fileAttr.isRegularFile())
+            .filter(file -> file.getFileName().toString().endsWith(".sol"))
+            .forEach(file -> analyzeFile(file.toFile()));
+        } catch (IOException e) {
+          e.printStackTrace();
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Use a logger to log this exception. ([squid:S1148](https://sonarcloud.io/organizations/stylianos-agapiou-sonarsource-gi/rules#rule_key=squid%3AS1148))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.sonarsource.solidity%3Asonar-solidity&pullRequest=29&issues=AWRlwlwkl6vpliCe6h0k&open=AWRlwlwkl6vpliCe6h0k)

[ISSUE_KEY]: AWRlwlwkl6vpliCe6h0k